### PR TITLE
Replace DriverWithContext.ExecuteQuery with generic ExecuteQuery standalone function

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Review by AndyHeap-NeoTech — COMMENTED** [review]
I think there should be information in the description around the transformers behaviours with retries. Does the interface require a callback that is invoked on a retry?

**Review by AndyHeap-NeoTech — APPROVED** [review]
Everything looks ok to me.

File: neo4j/driver_with_context.go [hunk 1]
```diff
@@ -382,66 +352,187 @@ func (d *driverWithContext) Close(ctx context.Context) error {
...
+//				Keys:    keys,
+//				Records: records,
+//				Summary: summary,
+//			}, nil
+//		})
+//		eagerResult := result.(*neo4j.EagerResult)
+//		// do something with eagerResult
+//
+// The available ResultTransformer implementation, EagerResultTransformer, computes an *EagerResult.
+// As the latter's name suggests, this is not optimal when the result is made of an important number of records.
```
Comment (AndyHeap-NeoTech) [inline neo4j/driver_with_context.go]: should this be 'made from a large number or records'?

File: neo4j/driver_with_context.go [hunk 2]
```diff
@@ -382,66 +352,187 @@ func (d *driverWithContext) Close(ctx context.Context) error {
...
+// In that situation, it is advised to create a custom implementation of ResultTransformer APIs, which do not
+// impose to keep all records in memory.
```
Comment (AndyHeap-NeoTech) [inline neo4j/driver_with_context.go]: 'which do not require keeping all records in memory'

Comment (fbiville) [reply neo4j/driver_with_context.go]: thanks fixed in https://github.com/neo4j/neo4j-go-driver/pull/435/commits/016a0d2ee5b4fd4d30a291d489d8680fe256f53c

File: neo4j/driver_with_context.go (hunk 1 above)
Comment (fbiville) [reply neo4j/driver_with_context.go]: thanks fixed in https://github.com/neo4j/neo4j-go-driver/pull/435/commits/016a0d2ee5b4fd4d30a291d489d8680fe256f53c

**Review by AndyHeap-NeoTech — APPROVED** [review]

**Pull request merged**